src/clang-cpp-frontend/clang_cpp_language.cpp

```diff
@@ -95,3 +100,30 @@ bool clang_cpp_languaget::from_type(
   code = type2cpp(type, ns);
   return false;
 }
+
+extern "C"
+{
+#include <optimized_includes/cpp_includes.h>
+#undef ESBMC_FLAIL
+}
+
+const std::string &clang_cpp_languaget::esbmc_cpp_includes()
+{
+  // Dump clang headers into a temporary directory
+  static bool dumped = false;
+  /* About the path being static:
+   * The static member 'dumped' above is used to check whether the headers were
+   * ever extracted before. This guarantees that the same path is used
+   * during a run. And no more than one is required anyway */
+  static auto p =
+    file_operations::create_tmp_dir("esbmc-headers-%%%%-%%%%-%%%%");
```

Comment:
Would it be possible to give this a less generic prefix like `esbmc-cpp-headers` or similar?

Reply:
Done